src/test/kotlin/org/jmailen/gradle/kotlinter/functional/PureKotlinFunctionalTest.kt

```diff
@@ -0,0 +1,106 @@
+package org.jmailen.gradle.kotlinter.functional
+
+import com.mkobit.gradle.test.assertj.GradleAssertions.assertThat
+import com.mkobit.gradle.test.kotlin.testkit.runner.build
+import com.mkobit.gradle.test.kotlin.testkit.runner.buildAndFail
+import com.mkobit.gradle.test.kotlin.testkit.runner.info
+import com.mkobit.gradle.test.kotlin.testkit.runner.profile
+import com.mkobit.gradle.test.kotlin.testkit.runner.setupProjectDir
+import com.mkobit.gradle.test.kotlin.testkit.runner.stacktrace
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+
+class PureKotlinFunctionalTest {
```

Comment:
nit: prefer test names which describe what's under test as opposed to how it's tested. How about `KotlinterPluginTest`? Open to switching to something like `kotlintest` if having nested context setups for JVM and Android seems useful.